packages/data-stores/src/site/types.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * External dependencies
+ */
+
+import { Action } from 'redux';
+
+export const enum ActionType {
+	CREATE_SITE = 'CREATE_SITE',
+	RECEIVE_NEW_SITE = 'RECEIVE_NEW_SITE',
+	RECEIVE_NEW_SITE_SUCCESS = 'RECEIVE_NEW_SITE',
+	RECEIVE_NEW_SITE_FAILED = 'RECEIVE_NEW_SITE_FAILED',
+	FETCH_NEW_SITE = 'FETCH_NEW_SITE',
```

Comment:
What's the difference between `FETCH_` and `CREATE_`? 

I'm just wondering, mostly out of ignorance, why we'd need `fetchNewSite()` at all when we have the `createSite()` action.

Is for the `isFetching` selector?

Reply:
Thanks, that's a good question! It's (currently) just used for the `isFetching` selector so we can manage a loading state in the UI. I'm not fully certain of how `@wordpress/data` deals with controls, but from Redux DevTools it looks like the CREATE_SITE action never hits the reducer. Taking a walk through the [`redux-routine`](https://developer.wordpress.org/block-editor/packages/packages-redux-routine/) code in Gutenberg, it looks to me like for this Redux middleware, if it encounters an action type for which we have a control (`CREATE_SITE`), then [the control is called](https://github.com/wordpress/gutenberg/blob/5920ab8620e0a5ea6ed810e399dc42fd94d99abd/packages/redux-routine/src/runtime.js#L26), but the action isn't then dispatched as well.

So I _think_ this means that the middleware effectively blocks the action at that point to perform whatever we want it to do in the control (in this case make an API call), but it doesn't hit the reducer. If it encounters actions for which we don't have a matching control, [they're dispatched as normal](https://github.com/wordpress/gutenberg/blob/5920ab8620e0a5ea6ed810e399dc42fd94d99abd/packages/redux-routine/src/runtime.js#L40). To set a fetching state, we then need an additional action for that (which I've called here `FETCH_NEW_SITE` but maybe it needs to be a bit more descriptive, because "fetch" sounds like we're retrieving something rather than submitting a POST request).

Unless of course there's a solution in using `isResolving` instead for the `isFetching` state. I haven't quite worked out how `isResolving` works, either, so I'm curious to learn about that, too!

And of course please let me know if I'm way off on my understanding of this! 😄